### PR TITLE
Cognito user pool

### PR DESCRIPTION
**Pull request by SAPessi**
Merge support for Cognito User Pool authorizers (Issue #24)

### CODE REVIEW

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/model/CognitoAuthorizerClaims.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    private String tokenUse;
+    @JsonProperty(value = "cognito:username")
+    private String username;
+    private String email;
+    @JsonProperty(value = "email_verified")
+    private boolean emailVerified;
+    @JsonProperty(value = "auth_time")
+    private Long authTime;
+    private String exp;
+    private String iat;
```
Comment (jt0) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/model/CognitoAuthorizerClaims.java]: Provide friendly names for "exp" and "iat" like you did for the others?

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletRequest.java [hunk 2]
```diff
@@ -280,6 +280,7 @@ protected String generateQueryString(Map<String, String> parameters) {
         if (headerValue == null) {
             return values;
         }
+        int entryCounter = 1;
```
Comment (jt0) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletRequest.java]: What is this for?

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/model/CognitoAuthorizerClaims.java (hunk 1 above)
Comment (SAPessi) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/model/CognitoAuthorizerClaims.java]: I'm providing friendly methods to get the parsed ZonedDateTime called `getExpiration` and `getIssuesAt`, that's why I left the raw fields with the same name as the original data

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletRequest.java (hunk 2 above)
Comment (SAPessi) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletRequest.java]: Not used, forgot there from when I was using the counter to name the key if there was no other name specified. Good catch. Removed.

**Pull request merged**